ccf_transaction_processor/transaction_processor/pdo_tp.cpp

```diff
@@ -19,129 +19,131 @@ using namespace std;
 using namespace ccf;
 using namespace tls;
 
-
 namespace ccfapp
 {
 
-    TPHandlerRegistry ::TPHandlerRegistry (kv::Store& store):
-        UserHandlerRegistry(store),
-        enclavetable(store.create<string, EnclaveInfo>("enclaves")),
-        contracttable(store.create<string, ContractInfo>("contracts")),
-        ccltable(store.create<string, ContractStateInfo>("ccl_updates")),
-        signer(store.create<string, map<string, string>>("signer"))
-    {
-        ledger_signer_local = NULL;
-    }
-
     string TPHandlerRegistry ::sign_document(const string& document){
         vector<uint8_t> doc_vector(document.begin(), document.end());
         auto sign = ledger_signer_local->sign(doc_vector);
         return b64_from_raw(sign.data(), sign.size());
     }
 
-    void TPHandlerRegistry::init_handlers(kv::Store& store){
+    string TPHandlerRegistry ::vector_to_string(const vector<uint8_t>& vec){
+        string s;
+        for (auto v : vec){
+            s += to_string(v);
+        }
+        return s;
+    }
 
-        UserHandlerRegistry::init_handlers(store);
+    TPHandlerRegistry ::TPHandlerRegistry (AbstractNodeContext& context):
+        UserEndpointRegistry(context),
+        enclavetable("enclaves"),
+        contracttable("contracts"),
+        ccltable("ccl_updates"),
+        signer("signer")
+    {
+        ledger_signer_local = NULL;
 
         //======================================================================================================
         // ping handler implementation
-        auto ping = [this](kv::Tx& tx, const nlohmann::json& params) {
-            return make_success(true);
+        auto ping = [this](auto& ctx, const nlohmann::json&) {
+            return ccf::make_success(true);
         };
```

Comment:
i know everyone loves auto's in c++. but they can really confuse more than help.

Reply:
agreed, but here i am simply following their recommended template for CCF apps. note that tx has changed to ctx with the version change, and i am sure this is not the last time we should expect app inference changes.